packages/initiatives/src/migrations/upgrade-one-dot-one.ts

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+import { getProp, cloneObject } from "@esri/hub-common";
+import { IInitiativeModel, IInitiativeItem } from "@esri/hub-common";
+
+export function upgradeToOneDotOne(
+  model: IInitiativeModel,
+  portalUrl?: string
+): IInitiativeModel {
+  const curVersion = getProp(model, "item.properties.schemaVersion");
+  if (curVersion < 1.1) {
+    const clone = cloneObject(model) as IInitiativeModel;
+    // store the schemaVersion
+    clone.item.properties.schemaVersion = 1.1;
+    // add the assets...
+    _addDefaultResources(clone, portalUrl);
+
+    if (!clone.data.values.bannerImage) {
+      clone.data.values.bannerImage = {
+        source: "bannerImage",
+        display: {
+          position: { x: "50%", y: "10%" }
+        }
+      };
+    }
+    return clone;
+  } else {
+    // console.debug(`Not upgrading CurVersion: ${curVersion}`);
+    return model;
+  }
+}
+
+export function _addDefaultResources(
+  model: IInitiativeModel,
+  portalUrl?: string
+): IInitiativeModel {
+  if (!model.data.assets) {
+    model.data.assets = [
+      {
+        id: "bannerImage",
+        url: getResourceUrl(model.item.id, "detail-image.jpg", portalUrl),
+        properties: {
+          type: "resource",
+          fileName: "detail-image.jpg",
+          mimeType: "image/jepg"
+        },
+        license: {
+          type: "none"
+        }
+      },
+      {
+        id: "iconDark",
+        url: getResourceUrl(model.item.id, "icon-dark.png", portalUrl),
+        properties: {
+          type: "resource",
+          fileName: "icon-dark.png",
+          mimeType: "image/png"
+        },
+        license: {
+          type: "none"
+        }
+      },
+      {
+        id: "iconLight",
+        url: getResourceUrl(model.item.id, "icon-light.png", portalUrl),
+        properties: {
+          type: "resource",
+          fileName: "icon-light.png",
+          mimeType: "image/png"
+        },
+        license: {
+          type: "none"
+        }
+      }
+    ];
+  }
+  return model;
+}
+
+export function getResourceUrl(
+  itemId: string,
+  resourceName: string,
+  portal?: string,
+  folder?: string
+): string {
```

Comment:
i doubt i'm going to be able to convince you to pass a single POJO argument to methods like this with props so you can use methods like `getPortalUrl()` easily, so i'll just skip it.

Reply:
I *could* be convinced... but lets land this geezer... this function is marked private anyhow